rust/lance/src/index.rs

```diff
@@ -101,6 +103,12 @@ pub trait DatasetIndexExt {
         params: &dyn IndexParams,
         replace: bool,
     ) -> Result<()>;
+
+    async fn remap_index(
```

Comment:
Let's just make a `optimize_index()` method to contain this remapping, deletion, or later incremental update together?

From the user's aspect, it is hard for them to understand what impact of exact each action is.  An general `optimize_index()` for them would be "call this method and your index will be faster"

Reply:
The user would never call this method (they would not be able to provide `row_id_map`).  I see your point though. I will move this out of `DatasetIndexExt` and call straight into the lower level method later.